superset/db_engine_specs/presto.py

```diff
@@ -900,10 +902,19 @@ def _partition_query(cls, table_name, limit=0, order_by=None, filters=None):
                 l.append(f"{field} = '{value}'")
             where_clause = "WHERE " + " AND ".join(l)
 
+        presto_version = database.get_extra().get("version")
+
+        # Partition select syntax changed in v0.199, so check here.
+        # Default to the new syntax if version is unset.
+        partition_select_clause = (
+            f'SELECT * FROM "{table_name}$partitions"'
+            if not presto_version or presto_version >= "0.199"
```

Comment:
I think you may want something like 
```
version_tuple = lambda v: tuple(map(int, v.split('.')))
version_tuple(presto_version) > (0, 199)
```

Reply:
I would suggest using `distutils` which provides a version comparison method as described [here](https://stackoverflow.com/a/6972866).

Reply:
I would suggest using `distutils` which provides a version comparison method as described [here](https://stackoverflow.com/a/6972866).

Reply:
Although string compare is _probably_ going to work in almost all cases, I agree that using `distutils` is a better solution. Updated!